### PR TITLE
HBASE-23954 SplitParent region should not be balanced

### PR DESCRIPTION
**Pull request by nyl3532016**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 34s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m  7s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m  1s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  15m 58s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| -1 :x: |  findbugs  |   3m  0s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  44m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | dupname asflicense spotbugs findbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux 56e02120553d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-general-check/output/patch-findbugs-hbase-server.txt |
| Max. process+thread count | 93 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java [hunk 1]
```diff
@@ -581,9 +584,11 @@ public ServerName getRegionServerOfRegion(RegionInfo regionInfo) {
       for (ServerName serverName : onlineServers) {
         ServerStateNode serverNode = serverMap.get(serverName);
         if (serverNode != null) {
-          ensemble.put(serverNode.getServerName(), serverNode.getRegionInfoList().stream()
-            .filter(region -> !isTableDisabled(tableStateManager, region.getTable()))
-            .collect(Collectors.toList()));
+          ensemble.put(serverNode.getServerName(),
+            serverNode.getRegionInfoList().stream()
+                .filter(region -> !isTableDisabled(tableStateManager, region.getTable())
```
Comment (infraio) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java]: filter twice is easy to read?
.filter(region -> !isTableDisabled(tableStateManager, region.getTable()).filter(region -> !region.isSplitParent())

Comment (nyl3532016) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java]: ok , right

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 36s |  master passed  |
| +1 :green_heart: |  compile  |   1m  7s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  1s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 41s |  hbase-server in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  5s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 53s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   0m 39s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| -0 :warning: |  unit  |  70m 38s |  hbase-server in the patch failed.  |
|  |   |  99m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux b2ecc15ec268 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/testReport/ |
| Max. process+thread count | 9179 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 17s |  master passed  |
| +1 :green_heart: |  compile  |   1m  0s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  96m 54s |  hbase-server in the patch passed.  |
|  |   | 124m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3d7307a2aa2f 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/testReport/ |
| Max. process+thread count | 7055 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/1/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 18s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 18s |  master passed  |
| +0 :ok: |  spotbugs  |   4m 57s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| -1 :x: |  findbugs  |   3m 28s |  hbase-server in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 48s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  21m 37s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| -1 :x: |  findbugs  |   4m 30s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 21s |  The patch does not generate ASF License warnings.  |
|  |   |  55m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | dupname asflicense spotbugs findbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux feb166d26ff3 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-general-check/output/branch-findbugs-hbase-server.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-general-check/output/patch-findbugs-hbase-server.txt |
| Max. process+thread count | 84 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 37s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 15s |  master passed  |
| +1 :green_heart: |  compile  |   1m  3s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 34s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 46s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 16s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  98m 58s |  hbase-server in the patch passed.  |
|  |   | 128m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 84fbf37c76b4 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/testReport/ |
| Max. process+thread count | 7236 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 13s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 58s |  master passed  |
| +1 :green_heart: |  compile  |   1m 51s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 20s |  branch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   1m 11s |  hbase-server in master failed.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 21s |  patch has no errors when building our shaded downstream artifacts.  |
| -0 :warning: |  javadoc  |   1m  8s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| -0 :warning: |  unit  | 121m 31s |  hbase-server in the patch failed.  |
|  |   | 169m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1264 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 392d4d07c965 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 31484f007f |
| Default Java | 2020-01-14 |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-jdk11-hadoop3-check/output/branch-javadoc-hbase-server.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-jdk11-hadoop3-check/output/patch-javadoc-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/testReport/ |
| Max. process+thread count | 6817 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1264/2/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**